vrrpd/vrrp_vty.c

```diff
@@ -314,6 +294,7 @@ DEFPY(vrrp_autoconfigure,
 	return CMD_SUCCESS;
 }
 
+/* XXX: yang conversion */
```

Comment:
YANG has the capability to model defaults; VRRP and other daemons have commands to change certain defaults as well, local to themselves, but YANG defaults are fixed attributes of a leaf. I'm not sure how to properly model this either. My initial thought would be to make defaults leaves themselves, that then have yang defaults, but it seems like a corruption of the arguably cleaner model of just setting the default directly on the relevant node instead of adding a level of indirection. What do you think?

Reply:
Understood.

I think the simplest solution to this problem is to leave the VRRP YANG module as is and make the ability to change defaults a CLI-only feature. This is what you've done so far, but the `vrrp default` command is affecting vrrpd as a whole and not only the VRRP instances configured through the CLI. What you could do instead is this:
* Make `vrrp_vrouter_create()` always use the defaults from the YANG module (to honor the YANG contract);
* Change the `vrrp (1-255)` command to push different default values when `vrrp default` is configured.

This way `vrrp default` would affect only VRRP instances configured through the CLI.

Other option would be to do something similar to what the IETF BGP module does:
```
      BGP
        |
        +-> [ global BGP configuration ]
          +-> AFI / SAFI global
        +-> peer group
          +-> [ peer group config ]
          +-> AFI / SAFI [ per-AFI overrides ]
        +-> neighbor
          +-> [ neighbor config ]
          +-> [ optional pointer to peer-group ]
          +-> AFI / SAFI [ per-AFI overrides ]";
```

You could add a global VRRP container containing the default values, and allow VRRP instances to either inherit these global defaults or override them. Not sure if that's worth doing though, I tend to think the CLI option should be enough.